python/cudf/cudf/tests/test_stats.py

```diff
@@ -442,7 +449,34 @@ def test_corr1d(data1, data2, method):
     ps2 = gs2.to_pandas()
 
     got = gs1.corr(gs2, method)
-    expected = ps1.corr(ps2, method)
+
+    ps1_align, ps2_align = ps1.align(ps2, join="inner")
+
+    is_singular = (
+        len(ps1_align.dropna()) == 1 and len(ps2_align.dropna()) > 0
+    ) or (len(ps2_align.dropna()) == 1 and len(ps1_align.dropna()) > 0)
+    is_identical = (
+        len(ps1_align.dropna().unique()) == 1 and len(ps2_align.dropna()) > 0
+    ) or (
+        len(ps2_align.dropna().unique()) == 1 and len(ps1_align.dropna()) > 0
+    )
+
+    # Pearson correlation leads to division by 0 when either sample size is 1.
+    # Spearman allows for size 1 samples, but will error if all data in a
+    # sample is identical since the covariance is zero and so the correlation
+    # coefficient is not defined.
+    cond = (is_singular and method == "pearson") or (
+        is_identical and not is_singular and method == "spearman"
+    )
+    if method == "spearman":
+        import scipy
+
+        expected_warning = scipy.stats._warnings_errors.ConstantInputWarning
+    elif method == "pearson":
+        expected_warning = RuntimeWarning
+
+    with expect_warning_if(cond, expected_warning):
```

Comment:
I'm not so sure this is a major improvement over just:

```python
if cond:
    with pytest.warns(expected_warning):
```

My worry with helpers like this is they don't end up being used everywhere that they are meant to - because developers don't typically reach for a helper for something like this.

If an existing call to `warns()` needs to change to happen conditionally, a developer is _much_ more likely to just move it inside a new `if` and be done with it.

As Nick would say though, this is a strong opinion - weakly held. If you feel strongly otherwise, please feel free to keep this.

Reply:
The downside of not having this is that you have to duplicate the code executed within the context:
```
if cond:
    with pytest.warns():
        foo()
else:
    foo()
```
Ideally `foo` would always be a simple one-liner, but that is not necessarily the case either (although maybe we should require it to be, it's certainly better to scope exception/warning handling as tightly as possible). Would you prefer the code duplication?

Reply:
Ah, that's a good point. I'm good with keeping it then - although yes, things like `raises()` and `warns()` should be tightly scoped anyway